Main entity creation

mdryden opened the pull request:
Some comments for the review

ICollection properties on models are virtual to utilize lazy-loading (see: http://www.entityframeworktutorial.net/EntityFramework4.3/lazy-loading-with-dbcontext.aspx)

BrewClubContext overrides OnModelCreating method in order to have inherited objects use the "Table Per Type" strategy.  (see: http://weblogs.asp.net/manavi/inheritance-mapping-strategies-with-entity-framework-code-first-ctp5-part-2-table-per-type-tpt)

mdryden committed:
added entities

mdryden committed:
project file

mdryden committed:
Entity Objects Creation #32

bradpowell commented on src/Core/Data/BrewClubContext.cs:
```diff
@@ -6,21 +6,25 @@ namespace mbsoft.BrewClub.Data
+		protected override void OnModelCreating(DbModelBuilder modelBuilder)
+		{
+			modelBuilder.Entity<Article>().ToTable("Articles");
+			
```
Should we be adding similar lines for the other EF objects like Classifieds?

mdryden replied:
Yes.  I've added Classifieds and Recipes to the explicit table definitions.  PostedItem is not required as it will be added automatically by EF.

bradpowell commented on src/Core/Data/BrewClubTestingInitializer.cs:
```diff
@@ -0,0 +1,35 @@
+
+		protected override void Seed(BrewClubContext context)
+		{
+			// this is pretty bad-ass - EF will recognize when the database needs to be changed, and this initializer is used to make it add/drop
```
Not quite sure I fully grasp what this does.  Can you explain it to me at some point?

mdryden replied:
I'm going to leave this in, because it's not currently in use.  We can discuss/design it in a future issue.

bradpowell commented on src/Core/Data/ClassifiedAnswer.cs:
```diff
@@ -0,0 +1,13 @@
+	public class ClassifiedAnswer
+	{
+		public UserProfile Author { get; set; }
+	}
```
Should there be a string property containing the answer text?

mdryden replied:
Yes, added.

bradpowell commented on src/Core/Data/ClassifiedQuestion.cs:
```diff
@@ -0,0 +1,17 @@
+		public UserProfile Author { get; set; }
+
+		public ClassifiedAnswer Answer { get; set; }
+
```
Should there be a string property containing the question text?

mdryden replied:
Yes, added.

bradpowell commented on src/Core/Data/PostedItem.cs:
```diff
@@ -0,0 +1,21 @@
+		public DateTime? LastEdit { get; set; }
+
+		public string Title { get; set; }
+
```
Should there be an ID property?  The "Inheritance with EF Code First: Part 2 – Table per Type (TPT)" article seemed to indicate that an ID is required for the pattern.

mdryden replied:
Yes, added.

bradpowell commented on src/Core/Data/Recipe.cs:
```diff
@@ -0,0 +1,19 @@
+
+		public string Description { get; set; }
+
+		// alot more needs to be added here.
```
http://grammarist.com/spelling/a-lot-alot/

mdryden replied:
Corrected my typo.  Nazi.

bradpowell replied:
ha ha

bradpowell commented on src/Core/Data/RecipeReview.cs:
```diff
@@ -0,0 +1,22 @@
+		
+		public bool AuthorBrewedIt { get; set; }
+
+		public bool? AuthorLikedIt { get; set; }
```
What are your thoughts on having a rating vs a boolean?  For example someone could give it 3 out of 5 stars.  We could use those ratings to calculate an average brewed rating for a recipe.

mdryden replied:
Removed the rating property altogether, for now, until we decide on an approach.

mdryden committed:
Added context table names for Classifieds and Recipes.

Added "string Answer" to ClassifiedAnswer.cs

Added "string Question" to ClassifiedQuestion.cs

Added "int PostedItemId" to PostedItem.cs

Corrected spelling of "a lot" in Recipe.cs

Removed the "AuthorLikedIt" property in RecipeReview.cs.  We will need to discuss further how to proceed with reviews.

mdryden commented:
Assigned back to @bradpowell for followup review.

pudds commented on src/Core/Data/RecipeReview.cs:
```diff
@@ -0,0 +1,20 @@
+		public string Comments { get; set; }
+		
+		public bool AuthorBrewedIt { get; set; }
+			
```
Removed the rating boolean, for now at least.  We will discuss.

bradpowell commented on src/Core/Data/ClassifiedQuestion.cs:
```diff
@@ -0,0 +1,19 @@
+		public string Question { get; set; }
+
+		public ClassifiedAnswer Answer { get; set; }
+
```
I think this should have an ID.  Since this is an EF object that will get represented as a table in the DB it'll need a foreign key ID to link it to the parent Classified.

mdryden replied:
Add an ID

bradpowell commented on src/Core/Data/RecipeReview.cs:
```diff
@@ -0,0 +1,20 @@
+		
+		public bool AuthorBrewedIt { get; set; }
+			
+
```
I think it should have an ID and parent ID pointing back to the recipe it relates to.

mdryden replied:
Parent IDs aren't required, EF handles the relationship automatically when you use a class within another class.

bradpowell commented:
Assigned back to @pudds with new comments.

mdryden committed:
added ids to all EF objects that didn't already have one

mdryden commented:
Assigned back to @bradpowell.

I've added IDs to all EF models which don't inherit from PostedItem base class.

I think we should be ready to merge this one now.

bradpowell commented:
Merge it baby.  I imagine we probably want to delete the branch but we've never talked about it yet so I'll leave it.
